wazuh/wazuh-dashboard-plugins
# Fixed filter modal of bar chart of vulnerabilities dashboard

### Description
When a filter is applied through the horizontal bar chart of the vulnerability module, the involved field of the chart is displayed repeatedly.
This happened because to have bars of different colors it was necessary to have an aggregation that referenced the same field as the x-axis in the bar chart.
Consulting with the OpenSearch community, they told us that the way to generate a bar graph with those colors is correct, but that would generate the problem with the modal repeating the field involved in the graph.
To avoid the modal problem, it was decided to leave the bar chart in monochrome, without the need for that extra aggregation mentioned above. With this, when interacting with the graph, the corresponding filter is directly applied.
 
### Issues Resolved
- #6170 

### Evidence
#### Vulnerability detector dashboard
![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/43619595/cac76e3f-a4cb-4c89-b6ac-e69a1e6f138c)

#### Vulnerability detector dashboard with applied filter by clicking on a value of the bar chart visualization
![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/43619595/71325b6e-e910-41f5-937a-01316c4b1673)


### Test
Before remember to create the mapping of port `9200:9200` in `osd1` in the `dev.yml`

- Create the index (wazuh-states-vulnerabilities) in the manager and insert data and mapping (using [comment](https://github.com/wazuh/wazuh-dashboard-plugins/issues/5763#issuecomment-1779564103)). 
- Navigate to `Threat Hunting` -> `Vulnerability Detection` and check the "Most detected vulnerabilities" and "The most vulnerable endpoints" monochrome
- Click a bar chart visualization value and verify that the filter has been applied.


### Check List
